Source/loadsave.cpp

```diff
@@ -559,7 +559,7 @@ static void LoadMonster(LoadHelper *file, int i)
 	pMonster->_myoff = file->nextLE<Sint32>();
 	pMonster->_mxvel = file->nextLE<Sint32>();
 	pMonster->_myvel = file->nextLE<Sint32>();
-	pMonster->_mdir = file->nextLE<Sint32>();
+	pMonster->_mdir = (direction)file->nextLE<Sint32>();

```

Comment:
In C++11 I'd generally use `static_cast<direction>(file->nextLE<Sint32>())`

In C++17 I'd use `direction { file->nextLE<Sint32>() }` (from http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0138r2.pdf)

Reply:
c++17 looks much nicer

Reply:
Yeah, this is just me not remembering how to do the right one, and being too lazy to look it up.

Reply:
@glebm, how about a simple `direction(file->nextLE<Sint32>())`?

Reply:
Using the C++17 method seams to give me a lot of narrowing warnings:
```
/home/ajenbo/code/diablo/devilutionX/Source/loadsave.cpp: In function ‘void devilution::LoadPlayer(devilution::{anonymous}::LoadHelper*, int)’:
/home/ajenbo/code/diablo/devilutionX/Source/loadsave.cpp:313:49: warning: narrowing conversion of ‘file->devilution::{anonymous}::LoadHelper::nextLE<int>()’ from ‘int’ to ‘uint8_t’ {aka ‘unsigned char’} [-Wnarrowing]
  313 |  pPlayer->_pmode = PLR_MODE{file->nextLE<Sint32>()};

```